### ISSUE TITLE
Connect_DualMode_MultiAddressFamilyConnect_RetrievedEndPoints_Success - Assert ::1

### ISSUE THREAD

**Issue by VincentBu** [dotnet/runtime#55053]
Run: [runtime 20210701.106](https://dev.azure.com/dnceng/public/_build/results?buildId=1216814&view=ms.vss-test-web.build-test-results-tab&runId=36305732&resultId=157674&paneView=debug)

Failed test:
```
net6.0-Linux-Release-x64-Mono_release-(Debian.10.Amd64.Open)ubuntu.1604.amd64.open@mcr.microsoft.com/dotnet-buildtools/prereqs:debian-10-helix-amd64-bfcd90a-20200121150006

- System.Net.Sockets.Tests.ConnectSync.Connect_DualMode_MultiAddressFamilyConnect_RetrievedEndPoints_Success

```
**Error message:**
```
Assert.Equal() Failure
Expected: ::ffff:127.0.0.1
Actual:   ::1


Stack trace
   at System.Net.Sockets.Tests.Connect`1.<Connect_DualMode_MultiAddressFamilyConnect_RetrievedEndPoints_Success>d__5[[System.Net.Sockets.Tests.SocketHelperArraySync, System.Net.Sockets.Tests, Version=6.0.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51]].MoveNext() in /_/src/libraries/System.Net.Sockets/tests/FunctionalTests/Connect.cs:line 103
--- End of stack trace from previous location ---
```

**Comment by karelz** [comment]
Duplicate of #54677

**Comment by karelz** [comment]
Reopening as it still happens, even after fix attempt via PR #54681 on 6/29
Related to #54677

Failures 5/20-7/20 (incl. PRs):

Date | Failures
-- | --
6/6 | 1x Official run
6/7 | 2x PR
6/8 | 4x PR
6/9 | 1x PR
6/10 | 1x PR
6/11 | 1x PR
6/12 | 2x Official run
6/13 | 1x Official run
6/14 | 3x PR
6/15 | 2x PR
6/16 | 1x Official run + 2x PR
6/18 | 1x Official run + 1x PR
6/19 | 1x PR
6/21 | 4x PR
6/22 | 2x PR
6/23 | 1x PR
6/24 | 2x PR
6/26 | 1x PR
6/28 | 3x PR (double records)
6/29 | 1x Official run + 6x PR (some double records)
6/30 | 1x Official run + 2x PR
7/1 | 1x Official run + 1x PR
7/2 | 2x Official run + 1x PR
7/5 | 1x PR
7/6 | 2x PR
7/7 | 1x Official run + 2x PR
7/8 | 1x Official run + 3x PR
7/9 | 2x PR
7/10 | 1x Official run
7/11 | 1x Official run + 1x PR
7/12 | 1x PR
7/13 | 3x PR
7/14 | 7x PR
7/15 | 2x PR
7/16 | 2x PR
7/17 | 3x PR
7/19 | 1x Official run + 4x PR
7/20 | 1x Official run + 2x PR (incomplete day)

**Comment by karelz** [comment]
Test disabled on Linux in PR #56002
